metadata-service/health-servlet/src/main/java/com/datahub/health/controller/HealthCheckController.java

```diff
@@ -0,0 +1,71 @@
+package com.datahub.health.controller;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.concurrent.TimeUnit;
+import lombok.AllArgsConstructor;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/check")
+@Tag(name = "HealthCheck", description = "An API for checking health of GMS and its clients.")
+public class HealthCheckController {
+  @Autowired
+  @Qualifier("elasticSearchRestHighLevelClient")
+  private RestHighLevelClient elasticClient;
+  private Supplier<ResponseEntity<String>> memoizedSupplier;
+
+  public HealthCheckController () {
+    this.memoizedSupplier = Suppliers.memoizeWithExpiration(
+        this::getElasticHealth, 5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Checks the memoized cache for the latest elastic health check result
+   * @return The ResponseEntity containing the health check result
+   */
+  @GetMapping(path = "/elastic", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Where is the combined healthcheck endpoint. Specifically, the one that produces a common message with all of the component checks? The one which would be used by a readiness check.